enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,353 @@
+
+Among the plugins already available we can find one for [OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py).
+This plugin relies on the usage of `oc` command line tool. It gathers data from
+OpenShift API resources and collects logs for pods and kubelet. While
+MicroShift shares similarities with this plugin, it is not directly usable for
+several reasons:
+* Using `oc` tool means it needs to be installed in the host. As seen before,
+  MicroShift strives for lightness and this package is not installed. Doing so
+  needs ~125MB.
+* Retrieves specific OpenShift API resources that are not present in MicroShift.
+
+Since none of the available plugins fulfills MicroShift's needs, we will need a
+new one.
+
+Requirements for the `sos` plugin follow:
+* Not require usage of other tools than what is already present in the host.
+  Introduce any of the `oc` commands in `microshift` binary. `microshift`
+  already supports a `get` command.
+* Have `must-gather` format to use already existing debug tools.
+* Retrieve MicroShift specific resources only, not covered by other plugins.
+  These include cluster resources, MicroShift configuration, MicroShift
+  version and MicroShift logs. Some specifics, like ovnk might require an
+  additional plugin, as openshift does. Other components, such as CRI-O, have
+  their own plugins.
+* Provide a single command pre-configured for full report gathering.
+  In order to have a single and simple command to gather the report, provide a
+  profile configuration file for `sos` to run all relevant plugins in one shot.
+* Must not leak sensitive data: IPs, hostnames, passwords, certificates, etc.
+
+Existing debugging tools in use by CEE include [omg](https://github.com/kxr/o-must-gather)
+and [omc](https://github.com/gmeghnag/omc). These provide a CLI like experience
+in terms of getting information from a live cluster, ressembling `oc` commands,
+but coming from a must-gather report. Since they were born out of must-gather,
+they expect the same file structure layout. The must-gather file system
+hierarchy is driven by `oc adm inspect` commands, gathering cluster resources
+data that these tools read upon later. To achieve the same results there are
+several options:
+* Use `inspect` logic to import it into `microshift`. This means an extra
+  dependency and more rebase logic. Testing would be easier, as `inspect`
+  drives the format of debugging tools.
+* Have `sos report` plugin mimick the file system hierarchy. It may be seen as
+  implementing `inspect` in a different language and in terms of
+  `microshift get` commands plus heavy string manipulation logic. The amount of
+  `get` commands required to produce the same output than `inspect` is on the
+  heavy side, unless a single `get` retrieves everything and have yaml
+  postprocessing afterwards. Plugin becomes harder to maintain and to test.
+* Generate an entirely different report from `sos report` plugin and then have
```

Comment:
This would be a lot of custom work. Let's move it to the alternatives list below.